### PR TITLE
OAuth 2.8 credentials support in Juju 2.9

### PR DESCRIPTION
**Pull request by tlm**
OAuth 2.8 credentials support in 2.9.

This change updates the Juju facades to return Kubernetes credential that a pre 2.9 releases can understand. This has been done through a facade bump for all users of the CloudSpec facade.

## Checklist

 - [x] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed
 - [x] Comments answer the question of why design decisions were made

## QA steps

1. Deploy a Juju 2.8 controller with a new model and deploy a workload.
2. Upgrade the controller to this PR
3. Check that the workload agent and support components have no errors and are communicating with the Kubernetes api.
4. Check controller logs for errors

## Bug reference

https://bugs.launchpad.net/juju/+bug/1929910

### CODE REVIEW

File: apiserver/facades/controller/firewaller/firewaller.go [hunk 1]
```diff
@@ -65,6 +65,11 @@ type FirewallerAPIV6 struct {
 	*FirewallerAPIV5
 }
 
+// FirewallerAPIV7 provides access to the Firewaller v7 API facade.
+type FirewallerAPIV7 struct {
+	*FirewallerAPIV6
```
Comment (hpidcock) [inline apiserver/facades/controller/firewaller/firewaller.go:70]: Shouldn't V6 have V7 and V7 have the main implementation?

**Review by hpidcock — APPROVED** [review]
Just one issue.

File: apiserver/facades/controller/firewaller/firewaller.go (hunk 1 above)
Comment (tlm) [reply apiserver/facades/controller/firewaller/firewaller.go:70]: No as that is the pattern used in this facade. It doesn't jive with the other facade patterns but I opted for consistency in this one.

**Comment by tlm** [comment]
$$merge$$

**Pull request merged**